pages/api/teams/[teamId]/invite.ts

```diff
@@ -0,0 +1,148 @@
+    }
+
+    try {
+      const team = await prisma.team.findUnique({
+        where: {
+          id: teamId,
+        },
+        include: {
+          users: {
+            select: {
+              userId: true,
+              role: true,
+            },
+          },
+        },
+      });
+
+      // check that the user is admin of the team, otherwise return 403
+      const teamUsers = team?.users;
+      const isUserAdmin = teamUsers?.some(
+        (user) =>
+          user.role === "ADMIN" &&
+          user.userId === (session.user as CustomUser).id
+      );
+      if (!isUserAdmin) {
+        return res.status(403).json("Only admins can send the invitation!");
+      }
+
+      const token = randomUUID().toString();
+
+      const expiresAt = new Date();
+      expiresAt.setHours(expiresAt.getHours() + 24); // invitation expires in 24 hour
+      const invitation = await prisma.invitation.create({
+        data: {
+          email,
+          token,
+          expires: expiresAt,
+          teamId,
+        },
+      });
+
+      // send invite email
+      const sender = session.user as CustomUser;
+
+      sendTeammateInviteEmail({
+        senderName: sender.name || "",
+        senderEmail: sender.email || "",
+        teamName: team?.name || "",
+        teamId: team?.id || "",
+        token,
+        to: email,
+      });
+
+      return res.status(200).json("Invitation send!");
+    } catch (error) {
+      errorHanlder(error, res);
+    }
+  } else if (req.method === "GET") {
+    // GET /api/teams/:teamId/invite
+    const session = await getServerSession(req, res, authOptions);
+
+    const { token, teamId, email } = req.query as {
+      token: string;
+      teamId: string;
+      email: string;
+    };
+
+    if (!session) {
+      res.redirect(`/login?next=/api/teams/${teamId}/invite?token=${token}`);
+    }
+
+    if (email !== session?.user?.email) {
+      return res.status(400).json("You are not invited to the team");
+    }
+
+    try {
+      const userTeam = await prisma.userTeam.findFirst({
+        where: {
+          teamId,
+          userId: (session?.user as CustomUser).id,
+        },
+      });
+
+      if (userTeam) {
+        // User is already in the team
+        return res.redirect(`/documents`);
+      }
+
+      const invitation = await prisma.invitation.findUnique({
+        where: {
+          token,
+        },
+      });
+
+      if (!invitation) {
+        return res.status(400).json("Invalid invitation token");
+      }
+
+      const currentTime = new Date();
+      if (currentTime > invitation.expires) {
+        return res.status(400).json("Invitation link has expired");
+      }
+
+      await prisma.team.update({
+        where: {
+          id: teamId,
+        },
+        data: {
+          users: {
+            create: {
+              userId: (session?.user as CustomUser).id,
+            },
+          },
+        },
+      });
+
```

Comment:
After the user is added to the team successfully, we need to delete the invitation record from invitation table, otherwise if the user removes himself and then tries to be invited again, there's a server error.